### DISCUSSION TITLE
remember application window positions

### DISCUSSION THREAD

**Discussion by soloturn** [pop-os/cosmic-epoch#987] [Q&A]
wayland itself does not allow applications to position their windows. does this mean cosmic-comp would be the component responsible to store application windows position, or is it smithay, or would this be something related to cosmic-session? session, because i saw a [gnome extension storing sessions](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager). when having more windows open on big screens, or multiple screens, it would be very practical to have it next time again.

**Comment by bkis** [comment]
A lot of time passed since this question has been asked. And the windows still open in default/random/arbitrary positions and sizes.

**Comment by lemmings67** [comment]
Restoring last window position/size is a really useful feature that miss a lot at this time
